File: backend/src/main/java/arbremojo/backend/model/StudentList.java
package arbremojo.backend.model;

import arbremojo.backend.model.user.Student;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;

@Entity
@Table(name = "student_list")
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class StudentList {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "student_list_id")
    private Integer studentListId;

    @Column(name = "absent")
    @ColumnDefault("false")
    private boolean absent;

    @Column(name = "minutes_late")
    @ColumnDefault("0")
    private int minutesLate;

    @Column(name = "time_table_id", nullable = false)
    private Integer timeTableId;

    @Column(name = "student_id", nullable = false)
    private Integer studentId;

    @ManyToOne(fetch = FetchType.EAGER, targetEntity = Student.class)
    @JoinColumn(name = "student_id", insertable = false, updatable = false)
    private Student student;

    @ManyToOne(fetch = FetchType.EAGER, targetEntity = TimeTable.class)
    @JoinColumn(name = "time_table_id", insertable = false, updatable = false)
    private TimeTable timeTable;
}

File: frontend/src/model/student-list.ts
import {Justification} from "./justification";
import {TimeTable} from "./time-table";
import {Student} from "./user/student";

export class StudentList {
  studentListId: number | undefined;
  absent: boolean;
  minutesLate: number;

  timeTableId: number;
  studentId: number;

  justification: Justification | undefined;
  timeTable: TimeTable | undefined;
  student: Student | undefined;

  constructor(absent: boolean, minutesLate: number, timeTableId: number, studentId: number, studentListId?: number) {
    this.absent = absent;
    this.minutesLate = minutesLate;
    this.timeTableId = timeTableId;
    this.studentId = studentId;
    this.studentListId = studentListId;
  }

  static fromJson(jsonStudentList: StudentList): StudentList {
    let studentList = new StudentList(jsonStudentList.absent, jsonStudentList.minutesLate, jsonStudentList.timeTableId,
      jsonStudentList.studentId, jsonStudentList.studentListId);
    if(jsonStudentList.justification != undefined) studentList.justification = Justification.fromJson(jsonStudentList.justification);
    if(jsonStudentList.timeTable != undefined) studentList.timeTable = TimeTable.fromJson(jsonStudentList.timeTable);
    if(jsonStudentList.student != undefined) studentList.student = Student.fromJson(jsonStudentList.student);

    return studentList;
  }

  static initializeStudentLists(jsonStudentLists: StudentList[]): StudentList[] {
    let studentLists: StudentList[] = [];
    if(jsonStudentLists != undefined) {
      for(let jsonStudentList of jsonStudentLists) {
        studentLists.push(StudentList.fromJson(jsonStudentList));
      }
    }
    return studentLists;
  }
}
